Add Plugin Labels and Prepackaged Plugin Info (#3395)
mattermost/docs: source/administration/plugins.rst

.. image:: https://user-images.githubusercontent.com/915956/66891467-1b18eb80-ef9e-11e9-9de3-37a3c5899bd8.png

Plugin Labels
~~~~~~~~~~~~~~~~

Plugins in the Marketplace are labeled to make it easier for administrators to choose plugins that fit their company's security and risk policies if they do not allow for community plugins to be used. 

> **Mattermost Official Plugins**

Reviewer: In v5.20, at least, we're not labelling official plugins at all, given the "spam" this label would produce. Should we invert the language here instead?

Plugins identified as "Official" are owned and managed by the Mattermost development team and are available in an open-source licencing repository for review.  These plugins are supported by our customer assistance team in production environments. 

> **Community Plugins**

Reviewer: It's worth noting that there are no community nor beta plugins right now: should we defer the labelling documentation altogether until that changes?

Author: I just didn't want it to slip as a task and have the link leading to nothing.  I'm fine with pulling this back and publish later.

Upgrading Plugins
~~~~~~~~~~~~~~~~~

> Upgrade a plugin on demand when a new version becomes available.  New versions of plugins you have already installed will display a link to easily install the upgraded plugin.  Some plugin versions may have breaking changes, check the release notes if performing a major version change.

Reviewer 2: ```suggestion
Upgrade a plugin on demand when a new version becomes available. New versions of plugins that you have already installed will display a link to easily install the upgraded plugins. Some plugin versions may have breaking changes; please check the release notes if you are performing a major version change.
```

Upgrading Plugins
~~~~~~~~~~~~~~~~~

> Upgrade a plugin on demand when a new version becomes available.  New versions of plugins you have already installed will be shown and can be installed with a single click. 

Reviewer: Pedantically, two clicks are required to confirm the upgrade.

Author: lol, true.

  In versions prior to 5.16, go to **System Console > Plugins (Beta) > Configuration**.

Pre-packaged plugins
~~~~~~~~~~~~~~~~~~~
Mattermost ships with a number of pre-packaged plugins written and maintained by Mattermost. Instead of building these features directly into the product, you can selectively enable the functionality your installation requires. Install pre-packaged plugins from the Plugin Marketplace, even if your system cannot directly connect to the internet.

Prior to v5.20, pre-packaged plugins were installed by default and could not be uninstalled without manually modifying the `prepackaged_plugins` directory. Any pre-packaged plugins installed prior to v5.20 and left enabled on upgrade will remain installed, but can now be uninstalled.

> Customplugins

Reviewer 2: ```suggestion
Custom plugins
```

Pre-packaged Plugins
~~~~~~~~~~~~~~~~~~~

> Mattermost ships with a number of pre-packaged plugins written and maintained by Mattermost. Instead of building these features directly into the product, you can selectively enable the functionality your installation requires. Pre-packaged plugins can be removed as of 5.20 via the System Console, or by modifying the ``prepackaged_plugins`` directory in your Mattermost installation.

Reviewer: ```suggestion
Mattermost ships with a number of pre-packaged plugins written and maintained by Mattermost. Instead of building these features directly into the product, you can selectively enable the functionality your installation requires. Install pre-packaged plugins from the Plugin Marketplace, even if your system cannot directly connect to the internet.
```

> As of v5.20, pre-packaged plugins are no longer installed by default, but are available from the plugin marketplace. Installing such plugins from the plugin marketplace works even when your server cannot access the internet by instead installing the version shipped with the server.

Reviewer: ```suggestion
Prior to v5.20, pre-packaged plugins were installed by default and could not be uninstalled without manually modifying the `prepackaged_plugins` directory. Any prepackaged plugins installed prior to v5.20 and left enabled on upgrade will remain installed, but can now be uninstalled.
```

> Previously enabled, pre-packaged plugins will remain installed on upgrading to v5.20. Any disabled, pre-packaged plugins will not show up in the system console but may be installed from the plugin marketplace.

Reviewer: ```suggestion
```

Pre-packaged Plugins
~~~~~~~~~~~~~~~~~~~

> Mattermost ships with a number of pre-packaged plugins written and maintained by Mattermost. Instead of building these features directly into the product, you can selectively enable the functionality your installation requires. Pre-packaged plugins can be removed after 5.20 via the System Console, or by modifying the ``prepackaged_plugins`` directory in your Mattermost installation.

Reviewer: Should we say `as of 5.20` instead of `after 5.20`?

> As of 5.20 Mattermost now ships pre-downloaded plugins that will be visible in the PLugin marketplace even if your Mattermost server does not have external internat acess as in some secure settings.  Only Mattermost "official" plugins are shipped with the Mattermost package as a result, community plugins are not available in the Plugin Marketplace on servers without internet access.

Reviewer 2: ```suggestion
As of 5.20 Mattermost now ships pre-downloaded plugins that will be visible in the plugin marketplace even if your Mattermost server does not have external internet access as in some secure settings.  Only Mattermost "official" plugins are shipped with the Mattermost package as a result. Community plugins are not available in the Plugin Marketplace on servers without internet access.
```

> As of 5.20 Mattermost now ships pre-downloaded plugins that will be visible in the PLugin marketplace even if your Mattermost server does not have external internat acess as in some secure settings.  Only Mattermost "official" plugins are shipped with the Mattermost package as a result, community plugins are not available in the Plugin Marketplace on servers without internet access.

Reviewer: I'm hesitant to introduce the term `pre-downloaded` and confuse things with `pre-packaged`: they are one and the same. Could we rephrase to clarify along the lines that, "As of v5.20, pre-packaged plugins are no longer installed by default, but are available from the plugin marketplace. Installing such plugins from the plugin marketplace works even when your server cannot access the internet by instead installing the version shipped with the server."

> If you previously had some plugins that were pre-packaged on your server, but were not enabled at time of upgrading to 5.20+, they will no longer be installed your server.  As mentioned - those previously pre-packaged plugins will still be available to be installed on your server via the Plugin Marketplace. 

Reviewer 2: ```suggestion
If you previously had some plugins that were pre-packaged on your server, but were not enabled at time of upgrading to 5.20+, they will no longer be installed your server.  As mentioned, those previously pre-packaged plugins will still be available to be installed on your server via the Plugin Marketplace. 
```

> If you previously had some plugins that were pre-packaged on your server, but were not enabled at time of upgrading to 5.20+, they will no longer be installed your server.  As mentioned - those previously pre-packaged plugins will still be available to be installed on your server via the Plugin Marketplace. 

Reviewer: If we simplify as above, we can simplify this further to emphasize the positive: "Previously enabled, pre-packaged plugins will remain installed on upgrading to v5.20. Any disabled, pre-packaged plugins will not show up in the system console but may be installed from the plugin marketplace."
